mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/PoolingProfileTypeBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.declaration.type;
+
+import static org.mule.metadata.api.builder.BaseTypeBuilder.create;
+import static org.mule.metadata.api.model.MetadataFormat.JAVA;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_EVICTION_INTERVAL_MILLIS;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_ACTIVE;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_IDLE;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_WAIT;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MIN_EVICTION_MILLIS;
+import org.mule.metadata.api.ClassTypeLoader;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.builder.ObjectTypeBuilder;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.config.PoolingProfile;
+
+/**
+ * Creates instances of {@link MetadataType} which represent a {@link PoolingProfile}.
+ *
+ * Notice that this representation of the type matches how the pooling profile is exposed
+ * through the tooling and the DSL. It's not an exact match of what you'd get running the
+ * pooling profile through a {@link ClassTypeLoader}
+ *
+ * @since 1.0
+ */
+public final class PoolingProfileTypeBuilder extends InfrastructureTypeBuilder {
+
+  /**
+   * @return a new {@link MetadataType} representation of a {@link PoolingProfile}
+   */
+  public MetadataType buildPoolingProfileType() {
+    final ObjectTypeBuilder objectType = create(JAVA).objectType().id(PoolingProfile.class.getName());
```

Comment:
every type that is parsed using the `ExtensionObjectTypeHandler` is enriched with an `TypeAliasAnnotation`. To be consistent with the other types, we could add the Alias to hand-built types like this one (in this case it could be `PoolingProfile`)